libs/chain-signatures/contract/src/state/running.rs

```diff
@@ -36,6 +41,8 @@ pub struct RunningContractState {
     pub parameters_votes: ThresholdParametersVotes,
     /// Votes for proposals to add new domains.
     pub add_domains_votes: AddDomainsVotes,
+    // Votes for proposals to add new TEE code hashes.
+    pub proposed_code_hashes: CodeHashesVotes,
```

Comment:
What was the reason for adding this field to the `RunningContractState` and the `ProtocolContractState` in general?

I understand that there is a grace period of multiple days, during which participants can update their node. So it is fine to keep this completely separate from the ProtocolState and even allow voting on the hash when the network is in Initializing or Resharing state.

Reply:
@bowenwang1996 [said](https://docs.nearone.org/doc/integrate-tee-into-mpc-network-OWBOeCofNK?commentId=229f37dd-d80e-46a8-a9d6-13ef20820c13&resolved=) it makes more sense to only allow voting on the hash when we're in the `Running` state, so I just went with that. Not sure why we'd want voting in `Initializing` or `Resharing`. @bowenwang1996, what do you think?

Reply:
The reason is that it doesn't make sense to vote when we are in initializing or resharing state, since voting requires a stable quorum of participants, which is not the case in initializing or resharing state. This is also to avoid unnecessarily complicating the contract logic.

Reply:
I see. The issue with tracking the votes here is that it is [a breaking API change](https://github.com/Near-One/mpc/pull/376#pullrequestreview-2802234130).

We would need to change the node s.t. they can handle both contract versions simultaneously. We would need to update the nodes before updating the contract.

Alternatively, we could store the votes in `TeeState` and just require that the contract is in running state whenever a vote is submitted.